File: app/src/main/java/com/example/davidtarielashvili/room/MainActivity.java
package com.example.davidtarielashvili.room;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import com.example.davidtarielashvili.CurrentQuote;
import com.example.davidtarielashvili.QuoteDialog;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private TextView tvQuote;
    private CurrentQuote currentQuote;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvQuote = findViewById(R.id.tv_quote);

        String[] quotes = getApplicationContext().getResources().getStringArray(R.array.array_quotes);
        QuoteRepository repository = new QuoteRepository(getApplication());

        for (String strQuote : quotes) {
            repository.insert(new Quote(strQuote));
        }

        ImageButton addBtn = findViewById(R.id.add_btn);
        addBtn.setOnClickListener(view -> {
            QuoteDialog dialog = new QuoteDialog(this);
            dialog.addNewQuote(R.layout.dialog_layout);
        });

        QuoteDao quoteDao = QuoteRoomDatabase.getDatabase(getApplicationContext()).quoteDao();
        quoteDao.getAllQuote().observe(this, (List<Quote> quoteList) -> {
            Quote quoteLast = quoteList.get(quoteList.size() - 1);
            currentQuote = new CurrentQuote(quoteLast.getQuote(),quoteLast.getUid());
            tvQuote.setText(quoteLast.getQuote());
        });

        ImageButton nextBtn = findViewById(R.id.next_btn);
        nextBtn.setOnClickListener(view -> {
            Quote quoteLast = quoteDao.findByID(currentQuote.getUid());
            currentQuote = new CurrentQuote(quoteLast.getQuote(),quoteLast.getUid());
            tvQuote.setText(quoteLast.getQuote());
        });


    }

}

File: app/src/main/java/com/example/davidtarielashvili/room/MainActivity.kt
package com.example.davidtarielashvili.room

import android.arch.lifecycle.Observer
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageButton
import android.widget.TextView
import com.example.davidtarielashvili.room.di.component.AppComponent
import com.example.davidtarielashvili.room.di.component.DaggerAppComponent

import com.example.davidtarielashvili.room.di.module.AppModule
import com.example.davidtarielashvili.room.di.module.RoomModule
import javax.inject.Inject

 class MainActivity : AppCompatActivity() {

    private var tvQuote: TextView? = null
    private var currentQuote: CurrentQuote? = null

    @Inject
    lateinit  var quoteRepository: QuoteRepository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val component: AppComponent by lazy {
            DaggerAppComponent
                    .builder()
//                .appModule(AppModule(application))
                    .roomModule(RoomModule(application))
                    .build()
        }

        component.inject(this)

//        val quoteRepository: QuoteRepository? = component.quoteRepository()


        tvQuote = findViewById(R.id.tv_quote)

        val quotes = applicationContext.resources.getStringArray(R.array.array_quotes)

        for (strQuote in quotes) {
            quoteRepository?.insert(Quote(strQuote))
        }

        val addBtn = findViewById<ImageButton>(R.id.add_btn)
        addBtn.setOnClickListener { view ->
            val dialog = QuoteDialog(this,quoteRepository!!)
            dialog.addNewQuote(R.layout.dialog_layout)
        }

        quoteRepository?.allQuote?.observe(this, Observer { quoteList ->
            val quoteLast = quoteList!![quoteList.size - 1]
            currentQuote = CurrentQuote(quoteLast.quote, quoteLast.uid)
            tvQuote!!.text = quoteLast.quote

        })

        val nextBtn = findViewById<ImageButton>(R.id.next_btn)
        nextBtn.setOnClickListener { view ->
            val quoteLast = quoteRepository!!.findByID(currentQuote!!.uid)
            currentQuote = CurrentQuote(quoteLast.quote, quoteLast.uid)
            tvQuote!!.text = quoteLast.quote
        }

    }

}
